### ISSUE TITLE
New tab flashes/blinks in Dark Mode

### ISSUE THREAD

**Issue by otaviocc** [brave/brave-ios#3378]
### Description: 

New tab flashes/blinks in Dark Mode

### Steps to Reproduce 
  1. Enable Dark Mode (either in iOS or in-app)
  2. Open a New Tab (by pressing "+")
  3. Observe

**Actual result:** 

https://user-images.githubusercontent.com/139272/110187783-e2d7ae80-7e19-11eb-88a7-c16b963d9f36.MP4

**Expected result:**

* It shouldn't blink. When in light mode, the background should be light. When in dark mode, the background should be dark.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

* 100% reproducible

**Brave Version:** 

Version 1.23.1.

**Device details:** 

* iPhone Xs (14.4))

**Comment by srirambv** [comment]
Am not sure the issue is fixed. I don't see the dark theme for tab cells when in tab tray still get the white border on top. This is on iPhone XR (iOS 13.5)

https://user-images.githubusercontent.com/17010094/111478899-6356b300-8728-11eb-9946-c867accd8d9c.MP4

On iPhone 7+ I see a different issue when ta new tab is opened and then going back to tray only shows the last tab first and then brings other tabs into focus which is a bug in itself not sure if it got introduced with #3379

https://user-images.githubusercontent.com/17010094/111479408-da8c4700-8728-11eb-82fa-8ebd30917971.MP4

**Comment by srirambv** [comment]
@iccub can you confirm if the fix is part of 1.24 build? If not can we move the issue out of the milestone and fix it for next release? I still the white gradient on tab tray even when device is in dark mode (as noted in the video's above)

**Comment by iccub** [comment]
I forgot to reopen it sorry, i will reopen and fix the dynamic theming part pater

**Comment by kjozwiak** [comment]
@iccub we should create a new issue re: fixing the dynamic theming. Have we landed any code in `1.23.x` that's associated with this issue? If we landed code, we should close this and leave it in `1.23.x` and either label it `QA/Yes` (if applicable) or `QA/No`. If there's no code that landed or was reverted, we can leave this opened. Generally, when there's code already associated with a certain release, the issue should be closed and left in that milestone. Makes it hard to follow when an issue has code that landed in multiple releases before it's fixed.

**Comment by iccub** [comment]
Good point, im fine with making new issue for it @kjozwiak,
can you add it and close this ticket?

**Comment by kjozwiak** [comment]
> Good point, im fine with making new issue for it @kjozwiak,
> can you add it and close this ticket?

Sure 👍 I'll get one created in a bit here and we'll close this one off. @srirambv @iccub is there anything that needs to QA'd here still? Or should we label this as `QA/No` and QA the fix via the new issue?

**Comment by iccub** [comment]
You can test it without changing theme dynamically but via Brave Settings -> Appearance -> Theme, same STR as in the ticket
And in the followup ticket we could test remaining stuff, so changing theme via control center

**Comment by otaviocc** [comment]
@iccub @kjozwiak @srirambv there's this one for dynamic color https://github.com/brave/brave-ios/issues/2558

**Comment by otaviocc** [comment]
@iccub @kjozwiak @srirambv for the top bar on top of the cell (the one with the title and icon) it was decided to leave it light https://github.com/brave/brave-ios/pull/3379#pullrequestreview-606436124.

**Comment by otaviocc** [comment]
@iccub @kjozwiak @srirambv and for the tab tray, there's this one open https://github.com/brave/brave-ios/issues/2558

**Comment by otaviocc** [comment]
@iccub @kjozwiak @srirambv what my issue/PR fixes is the background color, which was always flashing light, no matter the theme style. I didn't change the tab bar because there's the issue I mentioned before, open for iPad (but obviously reproduced on iPhone as well).

**Comment by iccub** [comment]
seems to work now
